From d2b528bf63809f9f54a31eb0745dd55f79a4c08f Mon Sep 17 00:00:00 2001
From: Fabian Groffen <grobian@gentoo.org>
Date: Mon, 21 Dec 2020 13:17:11 +0100
Subject: [PATCH] [ld64, cmake] build libprunetrie as static archive

We currently don't install libs, so referencing libprunetrie.dylib from
strip is bound to result in a trap.  Since I can't see a reason to
distribute libprunetrie, and ld(64) seems to incorporate it as well,
just statically link strip against libprunetrie.

Signed-off-by: Fabian Groffen <grobian@gentoo.org>
---
 ld64/src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git ld64/src/CMakeLists.txt ld64/src/CMakeLists.txt
index 1be1a5e..0f6a0f4 100644
--- ld64/src/CMakeLists.txt
+++ ld64/src/CMakeLists.txt
@@ -76,4 +76,4 @@ include_directories("${CMAKE_CURRENT_SOURCE_DIR}/abstraction")
 add_subdirectory(ld)
 add_subdirectory(other)
 
-add_library(prunetrie other/PruneTrie.cpp)
+add_library(prunetrie STATIC other/PruneTrie.cpp)
-- 
2.38.1

